Feature/1/search area

dev-seomoon opened the pull request:
<img width="753" alt="스크린샷 2022-02-22 오후 9 25 04" src="https://user-images.githubusercontent.com/52448114/155131981-f1007d41-7d30-4a35-a779-14b5788876f9.png">

Search Area 컴포넌트 UI 부분 구현 & 스타일링했습니다. 

**컴포넌트 구성**
```jsx
<SearchArea>
  <Title />
  <SearchBar />
  <RecommendArea />
<SearchArea>
```

코드나 PR에 수정했으면 하는 부분 코멘트 남겨주시고
이해 안되는 부분이 있으시면 코멘트 남겨주셔도 되고 디스코드로 질문 주셔도 됩니다

resolved: #1

Commit:
:package: Add styled-components package

Commit:
:lipstick: Add SearchArea component UI & styles

SearchArea 컴포넌트 구현 및 스타일링 완료,
svg는 file-loader가 필요해서 일단 하드코딩해서 넣었습니다.

dev-seomoon commented:
검색어 추천 영역 UI는 따로 이슈 발급해서 구현하겠습니다

Commit:
:package: Add package @svgr/webpack

To use svg as a React Component

Commit:
:rotating_light: Fix Lint error

off eslint "no-undef" rule

Commit:
:art: Use svg as a React Component

SearchArea 컴포넌트 안에 하드코딩되어 있던 svg를 파일로 분리하고 React Component로 import해서 사용.

Commit:
:lipstick: Add RecommendArea UI & styles

검색어 추천 영역 UI 구현 및 스타일링

Commit:
:package: Add react-responsive package

Commit:
:lipstick: Update UI responsively

SearchArea, RecommendArea 반응형 구현

exxocism commented on src/App.jsx:
```diff
@@ -1,5 +1,22 @@
+  return (
+    <div>
+      <SearchArea />
+      <GlobalStyle />
```
`Recommend-Request`
코드 작성하느라 고생하셨습니다.
혹시 GlobalStyle이 아랫부분에 위치한 이유를 알 수 있을까요?
페이지 렌더링이 한번 더 발생할것 같은 생각이 들어 말씀 드립니다.
```suggestion
```
해당 부분은 Index.js로 옮기는 것은 어떨지 다른 분들의 의견은 어떠신가요?

exxocism commented on src/component/SearchArea/SearchArea.jsx:
```diff
@@ -0,0 +1,30 @@
+import RecommendArea from '../RecommendArea/RecommendArea.jsx';
+
+const SearchArea = () => {
+  const isPC = useMediaQuery({ query: '(min-width: 1040px)' });
```
`Question`
궁금해서 여쭈어 보는 내용입니다. 
PC와 모바일을 구분하는 가장 좋은 방법이 어떤 거라고 생각하시나요? 
저는 절대적인 폭을 계산하는것도 하나의 방법이지만,
가로의 세로의 비율을 확인하는 방법이 좀더 효과적인 방법이라고 생각 합니다.
보통 저는 이렇게 적용하는데요. 혹시 이렇게 지정하신 이유를 알 수 있을까요?
```suggestion
  const isPC = useMediaQuery({ query: '(max-aspect-ratio: 3/4)' });
```

dev-seomoon replied:
좋은 질문 감사합니다!

https://clinicaltrialskorea.com/ 
위 사이트에서 폭 1040px 기준으로 반응형 분기가 이루어지고 있는 점 참고해서 적용했습니다 
정확히 pc와 모바일에 따른 분기가 아닌데 제가 변수명을 좀 부적절하게 지은 것 같네요

그리고 가로와 세로 비율을 분기점으로 삼는 방법은 사용해본 적이 없는데, 
모바일/pc 분기 외에
데스크탑 화면에서 전체화면이 아닌 경우나 태블릿 화면에 대해서도 혹시 적용이 가능한건가요?

Commit:
:lipstick: Set webfont 'SpoaqHanSans' as global font

Lela12 commented on src/App.jsx:
```diff
@@ -1,5 +1,24 @@
 }
+
+const GlobalStyle = createGlobalStyle`
+  @import url(//spoqa.github.io/spoqa-han-sans/css/SpoqaHanSansNeo.css);
```
Google font만 사용해왔는데, github을 통해서 웹폰트를 적용하신걸까요?

Lela12 reviewed (commented):
Good: UI기능 깔끔하게 잘만드셔서 좋습니다 :)

Lela12 commented on src/App.jsx:
```diff
@@ -1,5 +1,24 @@
 }
+
+const GlobalStyle = createGlobalStyle`
+  @import url(//spoqa.github.io/spoqa-han-sans/css/SpoqaHanSansNeo.css);
```
Google font만 사용해왔는데, github을 통해서 웹폰트를 적용하신걸까요?

dev-seomoon replied:
https://clinicaltrialskorea.com/ 에서 Spoqa Han Sans 폰트 사용하고 있길래 
[spoqa-hans-sans 공식 홈페이지](https://spoqa.github.io/spoqa-han-sans/) 참고해서 웹 폰트로 적용했습니다!

dev-seomoon replied:
리뷰 감사합니다 :)

Lela12 reviewed (approved):
Good: UI기능 깔끔하게 잘 만드셔서 좋습니다 :)

Commit:
:rotating_light: Add .eslintignore

Eslint 에러 해결 방식 통일.
기존 .eslintrc.json에서 no-undef rule off하는 방식
-> .eslintignore에 webpack.config.js 추가하는 방식.

dev-seomoon committed:
Merge branch 'dev' into feature/1/search-area

Lee-ye-ji commented:
**Recommend-Request:  에러, 기능 미구현, 오동작은 없습니다. 다음 형태의 개선을 추천합니다만, 
반영하지 않아도 좋고 제안해봅니다** 
- styled-components는 [CSS-in-JS ](https://www.daleseo.com/react-styled-components/)의 대표적인 라이브러리로써, 그 점이 활용되지 못한 것 같아서 조금 아쉽습니다😥 
더 자세히 설명하자면  `~.styles.jsx`로 css에 대한 내용을 모아놓는 것보단 각각의 컴포넌트 아래에 css를 표시하는 게 코드를 한 눈에 볼 수 있습니다!
```jsx
import React from 'react';
import PropTypes from 'prop-types';
// import { RecommendStyled } from './RecommendArea.styles.jsx';
import SearchIcon from '../../assets/icon_search.svg';
import styled from 'styled-components';

const Recommend = ({ content }) => {
  return (
    <RecommendStyled>
      <SearchIcon />
      {content}
    </RecommendStyled>
  );
};

Recommend.propTypes = {
  content: PropTypes.string,
};

export default Recommend;

const RecommendStyled = styled.li`
  svg {
    margin-right: 12px;
  }
  display: flex;
  align-items: center;
  line-height: 2.6;
  cursor: pointer;
`;
```
- 좀 더 추가했으면 좋았겠다하는 부분은 색깔 팔레트를 만들어서 [styled-component의 themeProvider](https://wonit.tistory.com/366) 를 이용하거나 global로 색깔을 객체로 만들어서 사용하는 부분도 있었으면 좋겠다라는 생각을 하였습니다.
- 마지막으로는 `App.jsx`의 GlobalStyle을 별도의 파일로 만드는 게 더 좋을 것 같습니다!

지금까지 긴 글 읽어주셔서 너무 감사드리고 UI 잘 만들어주셔서 감사합니다!!!🙇‍♀️

yeji-tomato reviewed (approved):
**Recommend-Request:  에러, 기능 미구현, 오동작은 없습니다. 다음 형태의 개선을 추천합니다만, 
반영하지 않아도 좋고 제안해봅니다** 
- styled-components는 [CSS-in-JS ](https://www.daleseo.com/react-styled-components/)의 대표적인 라이브러리로써, 그 점이 활용되지 못한 것 같아서 조금 아쉽습니다😥 
더 자세히 설명하자면  `~.styles.jsx`로 css에 대한 내용을 모아놓는 것보단 각각의 컴포넌트 아래에 css를 표시하는 게 코드를 한 눈에 볼 수 있습니다!
```jsx
import React from 'react';
import PropTypes from 'prop-types';
// import { RecommendStyled } from './RecommendArea.styles.jsx';
import SearchIcon from '../../assets/icon_search.svg';
import styled from 'styled-components';

const Recommend = ({ content }) => {
  return (
    <RecommendStyled>
      <SearchIcon />
      {content}
    </RecommendStyled>
  );
};

Recommend.propTypes = {
  content: PropTypes.string,
};

export default Recommend;

const RecommendStyled = styled.li`
  svg {
    margin-right: 12px;
  }
  display: flex;
  align-items: center;
  line-height: 2.6;
  cursor: pointer;
`;
```
- 좀 더 추가했으면 좋았겠다하는 부분은 색깔 팔레트를 만들어서 [styled-component의 themeProvider](https://wonit.tistory.com/366) 를 이용하거나 global로 색깔을 객체로 만들어서 사용하는 부분도 있었으면 좋겠다라는 생각을 하였습니다.
- 마지막으로는 `App.jsx`의 GlobalStyle을 별도의 파일로 만드는 게 더 좋을 것 같습니다!

지금까지 긴 글 읽어주셔서 너무 감사드리고 UI 잘 만들어주셔서 감사합니다!!!🙇‍♀️

jinyongp commented on src/component/SearchArea/SearchArea.jsx:
```diff
@@ -0,0 +1,30 @@
+          <input type="text" placeholder="질환명을 입력해 주세요. " />
+          {!isPC && <SearchIcon />}
+        </SearchBar.Box>
+        {isPC && <SearchBar.Button type="button">검색</SearchBar.Button>}
```
**Good: (어떠한 부분이) 좋습니다**
styled-components를 이런 식으로 이용한 건 처음 보는데 좋네요!

jinyongp commented on src/component/SearchArea/SearchArea.styles.jsx:
```diff
@@ -0,0 +1,79 @@
+    background-color: transparent;
+    line-height: 1.5;
+    font-size: 100%;
+    line-height: 1.15;
```
line-height 스타일이 중복되서 확인 부탁드려요.

jinyongp commented on src/App.jsx:
```diff
@@ -1,5 +1,24 @@
 import React from 'react';
+import SearchArea from './component/SearchArea/SearchArea.jsx';
```
**Major-Must**: js,jsx 확장자는 붙이지 않도록 규칙 업데이트했습니다. upstream 레포 pull 후 코드 수정하여 커밋 부탁드립니다.

dev-seomoon replied:
수정완료했습니다

ye-yo reviewed (approved):
**Good**: 깔끔하게 작성해주셔서 보기 좋았어요! 동적으로 여러 가지 텍스트가 나타날 수 있는 부분도 놓치지 않게 주석으로 깔끔하게 남겨주신 것 같아요!

src/component/RecommendArea/RecommendArea.jsx
```js
 {/* 추천 검색어 or 검색어 없음*/}
 <Info>추천 검색어</Info>
```
여기에 추가하자면 추후에 API 호출 기능이 완료될 경우 "검색중" 문구도 나타나야 할 것 같아요.
이 부분은 선경님께서 굳이 수정할 필요는 없을 것 같아 모두와 공유하기 위해서 기록 남깁니다!

jeonghun10 commented:
**Good**:UI구현이 깔끔하게 잘 이루어진것 같습니다!

jinyongp commented on src/App.jsx:
```diff
@@ -1,5 +1,24 @@
 export default function App() {
-  return <h1>Hello, World!</h1>;
+  return (
+    <div>
```
Recommend-Must: `GlobalStyle`의 경우 실제 그려질 컴포넌트가 아니니 의미없는 div를 사용하기 보다 [React Fragments](https://ko.reactjs.org/docs/fragments.html#short-syntax)를 이용하는게 좋을 것 같습니다.

dev-seomoon replied:
수정했습니다 좋은 제안 감사합니다

Commit:
:coffin: Remove dead style

line-height 스타일 중복 제거

Commit:
Merge branch 'dev' of https://github.com/jinyongp/wanted-codestates-project-7-10 into feature/1/search-area

Commit:
Merge branch 'feature/1/search-area' of https://github.com/dev-seomoon/wanted-codestates-project-7-10 into feature/1/search-area

Commit:
:rotating_light: Fix lint errors

웹팩 설정 변경으로 import 시 jsx 파일 확장자를 생략해야 한다는 error 발생 -> 확장자 생략으로 fix

Commit:
:art: Refactor CSS-in-JS structure

Commit:
:truck: Modify App.styles extension from jsx to js

jsx 코드가 사용된 부분이 없어 js 확장자로 변경

dev-seomoon commented:
@Lee-ye-ji 
- 스타일드 컴포넌트 적용 구조 개선에 대한 의견 반영하여, 
각각의 컴포넌트 아래에 스타일 작성했습니다. 
- `App.jsx` GlobalStyle 별도 파일로 분리했습니다. 

@jinyongp 
- 웹펙 설정 변경으로 lint 에러 발생해서
import 시 .jsx 확장자 들어가는 부분 전부 수정했습니다 (.jsx 확장자 제거)
- css 스타일 중복 (line-height) 제거했습니다


코드 수정 후 개발 서버에서 UI 정상적으로 그려지는 거 확인했습니다. 

이외에 제안해주신 부분이나 코멘트 남겨주신 부분은 추후 적용하도록 하겠습니다
리뷰 감사합니다.!

jinyongp reviewed (approved):
빠른 수정 감사합니다!

Commit:
:art: Remove meaningless div in App.jsx

exxocism reviewed (approved):
코멘트에 특별한 의견 없으시면 그냥 진행하셔도 될것 같습니다.

dev-seomoon commented:
@exxocism 우선 머지 진행하고 제안해주신 사항 좀 더 검토해보겠습니다 리뷰 감사합니다!
